cpp/src/arrow/array/validate.cc

```diff
@@ -294,6 +294,16 @@ struct ValidateArrayImpl {
 
 }  // namespace
 
+bool IsNullType(const DataType& type) {
```

Comment:
Is such a helper function the best approach? And if so, is it fine to just put it here?

Reply:
We could perhaps expose a `storage_type_id` method in `DataType`.

Reply:
Alternative could be to check the type's layout and check if `type.layout().buffers[0].kind == DataTypeLayout::ALWAYS_NULL` ?

Reply:
> We could perhaps expose a storage_type_id method in DataType.

And which returns itself if it's not an extension type?

Reply:
Right. cc @bkietz for opinions.

Reply:
I tried it now with a `DataType::storage_id()`